src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -36,29 +36,32 @@
   --force               Force overwrite of existing job directory
   --source_file_ext SOURCE_FILE_EXT
                         Select files from source data directory that match the given extension.
+  --threads_per_tool_run THREADS_PER_TOOL_RUN
+                        Number of threads to execute a single tool command. (default: 32)
                         
-Note: Based on our experiment with GTDB-Tk, we have determined that the optimal chunk size is 1000 genomes.
-With 4 batches running in parallel, each using 32 cores, it takes around 50 minutes to process a single chunk.
-To reflect this, we have set the "threads" and "program_threads" parameters in "_create_task_list" to 32, 
-indicating that each batch will use 32 cores. We have also set the execution time for GTDB-Tk to 65 minutes, 
-with some additional buffer time. To allow for a sufficient number of batches to be run within a given time limit, 
-we have set the "NODE_TIME_LIMIT_DEFAULT" to 5 hours. With these settings, we expect to be able to process up to 16 batches, 
-or 16,000 genomes, per node within the 5-hour time limit. We plan to make these parameters configurable based on 
-the specific tool being used. After conducting performance tests, we found that utilizing 32 cores per batch and 
-running 4 batches in parallel per NERSC node resulted in optimal performance, despite each node having a total of 
-256 cores.
 '''
 
 TOOLS_AVAILABLE = ['gtdb_tk', 'checkm2', 'microtrait', 'mash', 'eggnog']
 
-# estimated execution time (in minutes) for each tool to process a chunk of data
+NODE_TIME_LIMIT_DEFAULT = 5  # hours  # TODO: automatically calculate this based on tool execution time and NODE_THREADS
+# Used as THREADS variable in the batch script which controls the number of parallel tasks per node
+TASKS_PER_NODE_DEFAULT = 1
+
+# Task metadata - tool specific parameters for task generation and execution
+# chunk_size is the quantity of genomes processed within each task (default is 5000)
+#    for batch genome tools, such as gtdb_tk and checkm2, the chunk_size specifies the number of genomes grouped
+#    together for processing in a batch by the tool
+#    for single genome tools, such as microtrait and mash, the chunk_size is the number of genomes to process in a
+#    serial manner
+# exe_time is the estimated execution time for a single task (default is 60 minutes)
+# tasks_per_node is the number of parallel tasks to run on a node (default is 1)
+# node_time_limit is the time limit for the node we reserved for the task (default is 5 hours)
+# if no specific metadata is provided for a tool, the default values are used.
 TASK_META = {'gtdb_tk': {'chunk_size': 1000, 'exe_time': 65, 'tasks_per_node': 4},
```

Comment:
Will slurm keep running jobs on a node until it runs out of time?

Reply:
I think once jobs are all finished, the node will terminate.

Reply:
:+1: